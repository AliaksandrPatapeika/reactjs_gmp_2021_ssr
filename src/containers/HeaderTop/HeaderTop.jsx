import './HeaderTop.less';

import {useRouter} from 'next/router';
import PropTypes from 'prop-types';
import React, {useEffect, useRef} from 'react';
import {useDispatch, useSelector} from 'react-redux';

import {closeMovieDetails, showModal} from '../../actions/movies';
import Button from '../../components/Button';
import Logo from '../../components/Logo';
import {getQueryString} from '../../utils';

const HeaderTop = ({hideButton}) => {
  const activeMovieDetailsMovie = useSelector((state) => state.movie.activeMovieDetailsMovie);
  const query = useSelector((state) => state.query);
  const dispatch = useDispatch();
  const router = useRouter();
  const returnToMainButtonRef = useRef();

  const onClickHandler = () => {
    dispatch(closeMovieDetails());
    router.push({
      pathname: '/search',
      search: getQueryString(query)
    });
  };

  const showAddMovieWindow = () => {
    dispatch(showModal('addMovie'));
  };

  useEffect(() => {
    if (activeMovieDetailsMovie) {
      returnToMainButtonRef.current.focus();
    }
  }, [activeMovieDetailsMovie]);

  return (
    <div className="headerTopContainer">
      <Logo />
      {!hideButton && (
      <>
        {activeMovieDetailsMovie ? (
          <Button
            className="returnToMainButton"
            onClick={onClickHandler}
            ref={returnToMainButtonRef}
          >
            <i className="fa fa-search fa-flip-horizontal" aria-hidden="true" />
          </Button>
        ) : (
          <Button
            className="addMovieButton"
            onClick={showAddMovieWindow}
            title="+ ADD MOVIE"
          />
        )}
      </>
      )}
    </div>
  );
};

HeaderTop.propTypes = {
  hideButton: PropTypes.bool
};

HeaderTop.defaultProps = {
  hideButton: false
};

export default HeaderTop;
