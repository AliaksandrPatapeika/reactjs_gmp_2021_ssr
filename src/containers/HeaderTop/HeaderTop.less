@import "../../less/variables";

.headerTopContainer {
	display: flex;
	justify-content: space-between;

	.returnToMainButton {
		cursor: pointer;
		padding: 3px 6px;
		border-radius: 2px;
		color: @red;
		font-size: 23px;
		background-color: transparent;

		&:hover,
		&:focus {
			color: @red-hover;
			-webkit-box-shadow: 0 0 0 1px @white;
			-moz-box-shadow: 0 0 0 1px @white;
			box-shadow: 0 0 0 1px @white;
		}
	}

	.addMovieButton {
		cursor: pointer;
		width: 184px;
		height: 48px;
		border-radius: 4px;
		color: @red;
		font-size: 22px;
		font-family: @NetflixSans-Medium;
		background-color: @gray;
		opacity: 0.8;

		&:hover,
		&:focus {
			color: @white;
			background-color: @red-hover;
			opacity: 1;
		}
	}
}
