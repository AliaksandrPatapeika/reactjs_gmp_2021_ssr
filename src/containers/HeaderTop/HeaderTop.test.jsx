import {head} from 'lodash';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import HeaderTop from './HeaderTop';

describe('HeaderTop', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        activeMovieDetailsMovie: mockMovie
      }
    };
    const store = mockStore(initialState);
    const headerTop = renderer
      .create(
        <Provider store={store}>
          <HeaderTop />
        </Provider>
      )
      .toJSON();

    expect(headerTop).toMatchSnapshot();
  });
});
