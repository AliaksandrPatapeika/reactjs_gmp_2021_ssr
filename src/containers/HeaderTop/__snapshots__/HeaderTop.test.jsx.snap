// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTop renders correctly 1`] = `
<div
  className="headerTopContainer"
>
  <div
    className="logoContainer"
  >
    <span
      className="netflix"
    >
      netflix
    </span>
    <span
      className="roulette"
    >
      roulette
    </span>
  </div>
  <button
    className="returnToMainButton"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    
    <i
      aria-hidden="true"
      className="fa fa-search fa-flip-horizontal"
    />
  </button>
</div>
`;
